src/libnl3/patch/0002-mpls-remove-nl_addr_valid.patch

```diff
@@ -0,0 +1,55 @@
+From c89d1a129f71d3d2f76e6cbadb11ef41d8941a73 Mon Sep 17 00:00:00 2001
+From: Ann Pokora <apokora@juniper.net>
+Date: Tue, 25 May 2021 18:10:04 -0700
+Subject: [PATCH 2/2] mpls remove nl_addr_valid
+
+Signed-off-by: Ann Pokora <apokora@juniper.net>
+---
+ lib/route/nexthop.c       | 8 --------
+ lib/route/nh_encap_mpls.c | 4 ----
+ 2 files changed, 12 deletions(-)
+
+diff --git a/lib/route/nexthop.c b/lib/route/nexthop.c
+index 7a9904c..ac0095e 100644
+--- a/lib/route/nexthop.c
++++ b/lib/route/nexthop.c
+@@ -351,10 +351,6 @@ int rtnl_route_nh_set_newdst(struct rtnl_nexthop *nh, struct nl_addr *addr)
+ {
+ 	struct nl_addr *old = nh->rtnh_newdst;
+ 
+-	if (!nl_addr_valid(nl_addr_get_binary_addr(addr),
```

Comment:
Instead of removing the validation, can we fix the arguments.

```
replace nl_addr_get_len(addr) with nl_addr_get_family(addr)
```

Reply:
@smahesh That still isn't valid.  Look at the code for nl_addr_valid() in libnl3.  It is looking for an ASCII string and validating the string format.  addr does not container a string,  The return from nl_addr_get_binary_addr() is not a string. 
 We would have to first convert addr to string.  It's pointless.  nl_addr_valid() is not used anywhere else in libnl3, so I do not know why someone decided to use it in these MPLS functions.

Reply:
nl_addr_valid() is using the route_nh_set which is not mpls specific. this is the IP functions. how can this function pass now?

Reply:
@lguohan RTA_NEWDST is specific to MPLS.  NEWDST contains the MPLS labelstack associated with the NH.